Source/WebCore/editing/MarkupAccumulator.cpp

```diff
@@ -336,26 +338,26 @@ String MarkupAccumulator::resolveURLIfNeeded(const Element& element, const Strin
 
 RefPtr<Element> MarkupAccumulator::replacementElement(const Node& node)
 {
-    if (!m_shouldIncludeShadowDOM || !node.isShadowRoot())
+    RefPtr shadowRoot = dynamicDowncast<ShadowRoot>(node);
+    if (!shadowRoot || !m_shouldIncludeShadowDOM)
```

Comment:
Might be nice to do the `!m_shouldIncludeShadowDOM` early return earlier, in order to avoid the dynamicDowncast<>() in some cases.